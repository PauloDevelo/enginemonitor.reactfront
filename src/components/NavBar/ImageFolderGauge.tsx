import React, { useState, useEffect } from 'react';
import { Progress } from 'reactstrap';

type Props = {
    storageSizeInMB: number,
    storageSizeLimitInMB: number
};

const getColorProgress = (size: number, sizeLimit: number):string => {
  const ratio = size / sizeLimit;

  if (ratio < 0.5) {
    return 'success';
  }
  if (ratio >= 0.5 && ratio < 0.75) {
    return 'warning';
  }

  return 'danger';
};

const formatMbInCurrentCulture = (sizeInMb: number):string => new Intl.NumberFormat(window.navigator.languages[0], { maximumSignificantDigits: 2 }).format(sizeInMb);

const ImageFolderGauge = ({ storageSizeInMB, storageSizeLimitInMB }:Props) => {
  const [color, setColor] = useState(getColorProgress(storageSizeInMB, storageSizeLimitInMB));

  useEffect(() => {
    setColor(getColorProgress(storageSizeInMB, storageSizeLimitInMB));
  }, [storageSizeInMB, storageSizeLimitInMB]);

  return (
    <Progress color={color} value={storageSizeInMB} max={storageSizeLimitInMB}>
      {formatMbInCurrentCulture(storageSizeInMB)}
      {' '}
      of
      {' '}
      {formatMbInCurrentCulture(storageSizeLimitInMB)}
      MB
    </Progress>
  );
};

export default React.memo(ImageFolderGauge);
