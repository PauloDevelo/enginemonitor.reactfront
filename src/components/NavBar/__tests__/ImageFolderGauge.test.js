import React from 'react';
import { mount } from 'enzyme';

// eslint-disable-next-line no-unused-vars
import localforage from 'localforage';

import ImageFolderGauge from '../ImageFolderGauge';

import ignoredMessages from '../../../testHelpers/MockConsole';

jest.mock('localforage');

describe('ImageFolderGauge', () => {
  beforeEach(() => {
  });

  beforeAll(() => {
    ignoredMessages.length = 0;
    ignoredMessages.push('Could not find required `intl` object.');
  });

  afterEach(() => {
  });

  const imageFolderStatus = [
    { storageSizeInMB: 35, storageSizeLimitInMB: 100 },
    { storageSizeInMB: 60, storageSizeLimitInMB: 100 },
    { storageSizeInMB: 85, storageSizeLimitInMB: 100 },
  ];

  describe.each(imageFolderStatus)('Render', ({ storageSizeInMB, storageSizeLimitInMB }) => {
    it(`should display the image gauge when ${storageSizeInMB} Mb are used over ${storageSizeLimitInMB}`, (done) => {
      // Arrange

      // Act
      const imageFolderGaugeWrapper = mount(<ImageFolderGauge storageSizeInMB={storageSizeInMB} storageSizeLimitInMB={storageSizeLimitInMB} />);

      // Assert
      expect(imageFolderGaugeWrapper).toMatchSnapshot();
      done();
    });
  });
});
