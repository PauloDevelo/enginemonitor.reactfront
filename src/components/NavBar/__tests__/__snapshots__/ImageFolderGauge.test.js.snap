// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageFolderGauge Render should display the image gauge when 35 Mb are used over 100 1`] = `
<Memo(ImageFolderGauge)
  storageSizeInMB={35}
  storageSizeLimitInMB={100}
>
  <Progress
    barStyle={Object {}}
    color="success"
    max={100}
    min={0}
    style={Object {}}
    tag="div"
    value={35}
  >
    <div
      className="progress"
      style={Object {}}
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={35}
        className="progress-bar bg-success"
        role="progressbar"
        style={
          Object {
            "width": "35%",
          }
        }
      >
        35
         
        of
         
        100
        MB
      </div>
    </div>
  </Progress>
</Memo(ImageFolderGauge)>
`;

exports[`ImageFolderGauge Render should display the image gauge when 60 Mb are used over 100 1`] = `
<Memo(ImageFolderGauge)
  storageSizeInMB={60}
  storageSizeLimitInMB={100}
>
  <Progress
    barStyle={Object {}}
    color="warning"
    max={100}
    min={0}
    style={Object {}}
    tag="div"
    value={60}
  >
    <div
      className="progress"
      style={Object {}}
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={60}
        className="progress-bar bg-warning"
        role="progressbar"
        style={
          Object {
            "width": "60%",
          }
        }
      >
        60
         
        of
         
        100
        MB
      </div>
    </div>
  </Progress>
</Memo(ImageFolderGauge)>
`;

exports[`ImageFolderGauge Render should display the image gauge when 85 Mb are used over 100 1`] = `
<Memo(ImageFolderGauge)
  storageSizeInMB={85}
  storageSizeLimitInMB={100}
>
  <Progress
    barStyle={Object {}}
    color="danger"
    max={100}
    min={0}
    style={Object {}}
    tag="div"
    value={85}
  >
    <div
      className="progress"
      style={Object {}}
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={85}
        className="progress-bar bg-danger"
        role="progressbar"
        style={
          Object {
            "width": "85%",
          }
        }
      >
        85
         
        of
         
        100
        MB
      </div>
    </div>
  </Progress>
</Memo(ImageFolderGauge)>
`;
